processing/src/main/java/org/apache/druid/query/dimension/ListFilteredDimensionSpec.java

```diff
@@ -88,7 +87,7 @@ public DimensionSelector decorate(final DimensionSelector selector)
   private DimensionSelector filterWhiteList(DimensionSelector selector)
   {
     final int selectorCardinality = selector.getValueCardinality();
-    if (selectorCardinality < 0 || (selector.idLookup() == null && !selector.nameLookupPossibleInAdvance())) {
+    if (selectorCardinality < 0 || !selector.nameLookupPossibleInAdvance()) {
```

Comment:
why did this have the condition that the idLookup is null before?

Reply:
I'm not sure why it was there before, but I removed it because `ForwardingFilteredDimensionSelector` should not be used when the lookup is not possible in advance no matter whether `idLookup` is null or not (see the constructor of `ForwardingFilteredDimensionSelector`).